File: src/pydjinni/generator/objc/it/record/tests/BaseRecordTests.m
#import <XCTest/XCTest.h>
#import "Helper.h"
#import "BaseRecord.h"

@interface BaseRecordTests : XCTestCase
@end

@implementation BaseRecordTests

- (void)testCppBaseRecord {
    BaseRecord* record = [Helper getCppBaseRecord];

    XCTAssertEqual(record.value, 42);
}

- (void)testObjcBaseRecord {
    BaseRecord* record = [BaseRecord baseRecord];
    BaseRecord* returned_record = [Helper getHostBaseRecord:record];
    XCTAssertEqual(returned_record.value, 42);
}

@end

File: src/pydjinni/generator/objc/it/swift/record/tests/BaseRecordTests.swift
import XCTest
import RecordTestSwift

class BaseRecordTests: XCTestCase {

    func testCppBaseRecord() {
        let record = Helper.getCppBaseRecord()
        XCTAssertEqual(record.value, 42)
    }

    func testObjcBaseRecord() {
        let record = BaseRecord()
        let returnedRecord = Helper.getHostBaseRecord(recordType: record)
        XCTAssertEqual(returnedRecord.value, 42)
    }
}
